source/_addonStore/dataManager.py

```diff
@@ -270,15 +270,21 @@ def _cacheInstalledAddon(self, addonData: AddonStoreModel):
 		if not addonData:
 			return
 		addonCachePath = os.path.join(self._installedAddonDataCacheDir, f"{addonData.addonId}.json")
-		with open(addonCachePath, 'w') as cacheFile:
+		with open(addonCachePath, 'w', encoding='utf-8') as cacheFile:
 			json.dump(addonData.asdict(), cacheFile, ensure_ascii=False)
 
 	def _getCachedInstalledAddonData(self, addonId: str) -> Optional[InstalledAddonStoreModel]:
 		addonCachePath = os.path.join(self._installedAddonDataCacheDir, f"{addonId}.json")
 		if not os.path.exists(addonCachePath):
 			return None
-		with open(addonCachePath, 'r') as cacheFile:
-			cacheData = json.load(cacheFile)
+		try:
+			with open(addonCachePath, 'r', encoding='utf-8') as cacheFile:
+				cacheData = json.load(cacheFile)
+		except Exception:
+			log.exception(f"Invalid cached installed add-on data")
+			if NVDAState.shouldWriteToDisk():
+				os.remove(addonCachePath)

```

Comment:
I have mixed feelings about removing the cache here. As opposed to the add-on store cache, we can't as easily throw out and fetch a new cache file. I'd be more inclined to keep the invalid file for debugging purposes until the add-on is updated or re-installed.

```suggestion
```